### PR TITLE
Make dashboard left side nav an unordered list

### LINKED ISSUE #14841: Dashboard left-side navigation should be a list
**Describe the bug**

The left-side navigation options on the dashboard page (desktop view) are correctly wrapped in a `<nav>` element but they should also be wrapped in a `<ul>`, each as individual `<li>` list items.

This helps screen reader users navigate between the items, knowing how many options are in the list and their current position in that list.

**To Reproduce**

- Using a screen reader visit the left-hand navigation on the `/dashboard` page
- Notice the number of items in the navigation is not announced

**Expected behavior**

- Items are wrapped in a list and the number of items is announced when using a screen reader

**Screenshots**

screenshot of the navigation panel

### PR DESCRIPTION
**Pull request by secondl1ght**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
As per issue #14841 I have made the dashboard left-side navigation an unordered list with the anchor elements as the list items. To improve quality of screen reading.

## Related Tickets & Documents
Closes #14841 

## QA Instructions, Screenshots, Recordings

I don't believe this change will need any new tests to be written.

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: If tests need to be written for this change then please let me know.
- [ ] I need help with writing tests

### CODE REVIEW

File: app/views/dashboards/_actions.html.erb [hunk 1]
```diff
@@ -1,70 +1,93 @@
 <nav class="hidden m:block" aria-label="Dashboards">
-  <a class="crayons-link crayons-link--block <%= "crayons-link--current" if params[:action] == "show" && (params[:which] == "organization" || params[:which].blank?) %>"
-    href="<%= dashboard_path %>"
-    <%= params[:action] == "show" && (params[:which] == "organization" || params[:which].blank?) ? ' aria-current="page"'.html_safe : "" %>>
-    Posts
-    <span class="crayons-indicator"><%= @user.articles_count %></span>
-  </a>
+  <ul>
```
Comment (aitchiss) [inline app/views/dashboards/_actions.html.erb:2]: One thing I didn't explicitly mention on the issue is that visually we don't wany any change as a result of adding the list. We use tailwind-style classes, and I _think_ you can achieve this with the suggestion below (but probably best to double check!)

```suggestion
  <ul class="list-none p-0">
```

**Review by aitchiss — COMMENTED** [review]
Thanks for taking this on! I've left one small suggestion to tweak the class names so that visually nothing changes 👍

**Comment by msarit** [comment]
@secondl1ght Hey there! Thank you for the PR! 👋🏾 
Please remember to [sign the CLA](https://github.com/forem/forem/pull/14887#issuecomment-932580271)!

File: app/views/dashboards/_actions.html.erb (hunk 1 above)
Comment (msarit) [reply app/views/dashboards/_actions.html.erb:2]: I tested @aitchiss's suggestion and it does work! 👍🏾

**Review by msarit — APPROVED** [review]
@secondl1ght This PR looks great, thanks for contributing.

I'm approving, assuming you will incorporate Suzanne's suggestion [here](https://github.com/forem/forem/pull/14887/files#r721318966).

**Comment by secondl1ght** [comment]
@aitchiss @msarit thanks so much for the help on this, this was my first time receiving comments back on a PR and committing the changes suggested so that was cool to see how it works! I believe everything should be ready to merge now. :)

**Review by rhymes — APPROVED** [review]
Great job, thank you @secondl1ght and welcome to Forem :)

**Pull request merged**